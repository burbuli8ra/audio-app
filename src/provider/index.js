import React, { createContext, useContext, useReducer } from 'react';
import activeArtistReducer from './reducers/activeArtistReducer';
import activeAudioIndexReducer from './reducers/activeAudioIndexReducer';
import activeTrackListReducer from './reducers/activeTrackListReducer';

export const AppContext = createContext();

export const initialState = {
  activeArtist: null,
  activeAudioIndex: null,
  activeTrackList: []
};

export const mainReducer = (state, action) => ({
  ...state,
  activeArtist: activeArtistReducer(state.activeArtist, action),
  activeAudioIndex: activeAudioIndexReducer(state.activeAudioIndex, action),
  activeTrackList: activeTrackListReducer(state.activeTrackList, action)
});

export const StateProvider = ({ reducer, initialState, children }) => (
  <AppContext.Provider value={useReducer(reducer, initialState)}>
    {children}
  </AppContext.Provider>
);

export const useAppContext = () => useContext(AppContext);