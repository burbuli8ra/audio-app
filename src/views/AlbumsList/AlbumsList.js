import React, { useEffect, useState } from 'react';
import { Loader } from '../../components';
import Styled from './styled';
import albumsMock from '../../__mocks__/albums.json'

const AlbumList = () => {
  const [isLoading, setIsLoading] = useState(true);
  const [albums, setAlbums] = useState([]);

  // @todo replace with actual API call
  useEffect(() => {
    const timer = setTimeout(() => {
      setAlbums(albumsMock);
      setIsLoading(false);
    }, 2000);

    return () => clearTimeout(timer);
  }, []);

  return (
    isLoading
      ? <Loader />
      : (
        <>
          <Styled.Heading>Recent Releases</Styled.Heading>
          <Styled.List>
          {albums.map(album => (
            <Styled.AlbumLink
              key={`album-${album.id}`}
              to={{
                pathname: `album/${album.id}`,
                state: { albumId: album.id }
               }}
            >
              <Styled.Cover>
                <img
                  src={album.cover}
                  alt={`Cover of "${album.title}" by ${album.artist}`}
                />
              </Styled.Cover>
              <Styled.Details>
                <Styled.Title>{album.title}</Styled.Title>
                <Styled.Artist>by <em>{album.artist}</em></Styled.Artist>
              </Styled.Details>
            </Styled.AlbumLink>
          ))}
          </Styled.List>
        </>
      )
  );
};

export default AlbumList;