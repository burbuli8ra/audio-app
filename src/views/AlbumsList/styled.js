import styled from 'styled-components';
import { Link } from 'react-router-dom';

const Cover = styled.div`
  border: 3px solid #28aa90;
  border-radius: 50%;
  height: 120px;
  width: 120px;
  overflow: hidden;

  > img {
    height: 100%;
    width: 100%;
  }
`;
Cover.displayName = 'Cover';

const Title = styled.div`
  color: #28aa90;
  font-size: 18px;
  font-weight: bold;
`;
Title.displayName = 'Title';

const Artist = styled.div`
  color: #404040;
  font-size: 14px;
`;
Artist.displayName = 'Artist';

const Details = styled.div`
  padding: 20px 0;
  margin-left: 20px;

  ${Title} {
    margin-bottom: 8px;
  }
`;
Details.displayName = 'Details';

const AlbumLink = styled(Link)`
  display: flex;
  text-align: left;
  text-decoration: none;
`;
AlbumLink.displayName = 'AlbumLink';

const List = styled.div`
  display: inline-flex;
  flex-direction: column;

  ${AlbumLink} {
    margin-bottom: 20px;

    &:last-of-type {
      margin-bottom: 0;
    }
  }
`;
List.displayName = 'List';

const Heading = styled.div`
  color: #28aa90;
  font-size: 24px;
  font-weight: bold;
  text-decoration: underline;
  text-transform: uppercase;
  margin-bottom: 20px;
`;
Heading.displayName = 'Heading';

export default {
  AlbumLink,
  Artist,
  Cover,
  Details,
  Heading,
  List,
  Title
}