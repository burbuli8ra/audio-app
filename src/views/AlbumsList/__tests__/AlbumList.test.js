import React from 'react';
import { act, render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { StateProvider, initialState, mainReducer } from '../../../provider';
import AlbumsList from '../AlbumsList';

describe('AlbumsList view', () => {
  let wrapper;

  const setup = () => (
    <StateProvider reducer={mainReducer} initialState={initialState}>
      <Router>
        <AlbumsList />
      </Router>
    </StateProvider>
  );

  test('should match snapshot', () => {
    jest.useFakeTimers();
    wrapper = render(setup());

    act(() => jest.runAllTimers());

    expect(wrapper).toMatchSnapshot();
  });
});