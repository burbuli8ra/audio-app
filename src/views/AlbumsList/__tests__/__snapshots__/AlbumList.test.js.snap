// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlbumsList view should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-fzqBZW dUqHN"
      >
        Recent Releases
      </div>
      <div
        class="sc-fzokOt dLEZtS"
      >
        <a
          class="sc-fznZeY domBZc"
          href="album/1"
        >
          <div
            class="sc-fzpans hnvDCT"
          >
            <img
              alt="Cover of \\"The Villain Way\\" by Bob Villain"
              src="https://www.instagram.com/p/BFmPMNpS7ig/media"
            />
          </div>
          <div
            class="sc-fznKkj eZUTMj"
          >
            <div
              class="sc-fzplWN cEdPvy"
            >
              The Villain Way
            </div>
            <div
              class="sc-fznyAO elWmgt"
            >
              by 
              <em>
                Bob Villain
              </em>
            </div>
          </div>
        </a>
        <a
          class="sc-fznZeY domBZc"
          href="album/2"
        >
          <div
            class="sc-fzpans hnvDCT"
          >
            <img
              alt="Cover of \\"Live in Athens\\" by The Pure"
              src="https://www.instagram.com/p/B0EUBPBoziQ/media"
            />
          </div>
          <div
            class="sc-fznKkj eZUTMj"
          >
            <div
              class="sc-fzplWN cEdPvy"
            >
              Live in Athens
            </div>
            <div
              class="sc-fznyAO elWmgt"
            >
              by 
              <em>
                The Pure
              </em>
            </div>
          </div>
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-fzqBZW dUqHN"
    >
      Recent Releases
    </div>
    <div
      class="sc-fzokOt dLEZtS"
    >
      <a
        class="sc-fznZeY domBZc"
        href="album/1"
      >
        <div
          class="sc-fzpans hnvDCT"
        >
          <img
            alt="Cover of \\"The Villain Way\\" by Bob Villain"
            src="https://www.instagram.com/p/BFmPMNpS7ig/media"
          />
        </div>
        <div
          class="sc-fznKkj eZUTMj"
        >
          <div
            class="sc-fzplWN cEdPvy"
          >
            The Villain Way
          </div>
          <div
            class="sc-fznyAO elWmgt"
          >
            by 
            <em>
              Bob Villain
            </em>
          </div>
        </div>
      </a>
      <a
        class="sc-fznZeY domBZc"
        href="album/2"
      >
        <div
          class="sc-fzpans hnvDCT"
        >
          <img
            alt="Cover of \\"Live in Athens\\" by The Pure"
            src="https://www.instagram.com/p/B0EUBPBoziQ/media"
          />
        </div>
        <div
          class="sc-fznKkj eZUTMj"
        >
          <div
            class="sc-fzplWN cEdPvy"
          >
            Live in Athens
          </div>
          <div
            class="sc-fznyAO elWmgt"
          >
            by 
            <em>
              The Pure
            </em>
          </div>
        </div>
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
